src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -345,23 +343,44 @@ private Schema convertFieldType(Field field) {
       // https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-avro#logical_types
       schema = Schema.decimalOf(38, 9);
     } else if (value == StandardSQLTypeName.STRUCT) {
-      List<Schema.Field> fields = field.getSubFields().stream()
-        .map(this::getSchemaField)
-        .collect(Collectors.toList());
-      schema = Schema.recordOf(field.getName(), fields);
+      FieldList fields = field.getSubFields();
+      List<Schema.Field> schemafields = new ArrayList<>();
+      for (Field f : fields) {
+        Schema.Field schemaField = getSchemaField(f, collector);
+        // if schema field is null, that means that there was a validation error. We will still continue in order to
+        // collect more errors
+        if (schemaField == null) {
+          continue;
+        }
+        schemafields.add(schemaField);
+      }
+      // do not return schema for the struct field if none of the nested fields are of supported types
+      if (!schemafields.isEmpty()) {
+        schema = Schema.recordOf(field.getName(), schemafields);
+      }
     } else {
-      // this should never happen
-      throw new InvalidStageException(String.format("BigQuery column '%s' is of unsupported type '%s'.",
-                                                    field.getName(), value));
+      collector.addFailure(String.format("BigQuery column '%s' is of unsupported type '%s'.", field.getName(), value),
+                           "Supported types are: " +
+                             "INT64, NUMERIC, FLOAT64, STRING, BOOLEAN, BYTES, ARRAY, TIMESTAMP, DATE, " +
+                             "TIME or STRUCT.");
     }
     return schema;
   }
 
-  private void validatePartitionProperties() {
+  private void validatePartitionProperties(FailureCollector collector) {
+    String project = config.getDatasetProject();
+    String dataset = config.getDataset();
+    String tableName = config.getTable();
+    Table sourceTable = BigQueryUtil.getBigQueryTable(project, dataset, tableName,
+                                                      config.getServiceAccountFilePath(), collector);
+    if (sourceTable == null) {
+      // Table does not exist
+      collector.addFailure(String.format("BigQuery table '%s:%s.%s' must exist.", project, dataset, tableName), null)
```

Comment:
This specific check wont get added twice because of immediate `throw collector.getOrThrowException()` statement. This is added to be more defensive and avoid npe if table is not found.

In general if duplicate failures are added, 2 different errors will be shown.